### PR TITLE
K8s add support of vault for secrets

### LINKED ISSUE #475: Cloud Secrets Management
As a security engineer,
I want to implement Cloud Secrets Manager for managing sensitive data like IPFS pinning API keys, passwords, and seed phrases, so that I can securely store, access, and manage secrets required by the Gateway stack.

Acceptance Criteria:

- Cloud Secrets Manager should be configured to store sensitive data such as API keys, passwords, and seed phrases.
- Secrets should be encrypted and access-controlled based on roles and permissions.
- The application should be able to retrieve secrets securely from Cloud Secrets Manager.
- README Documentation on how to manage secrets should be provided.

Note:
Development teams need to add and relate all necessary implementation subtasks (e.g., integrating AWS Secrets Manager with the application, setting up access controls, encrypting secrets) to this story for comprehensive tracking.

### PR DESCRIPTION
**Pull request by saraswatpuneet**
# Problem

As a developer I would like to pull secrets from vault secret manager and use those for services

- [x] Support optional vault
- [x] Update secrets to pull from vault hook
- [x] Test the vault integration

Closes #475

### CODE REVIEW

File: deployment/k8s/vault/vault_secret_store.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  provider:
+    vault:
+      server: "http://vault.vault.svc.cluster.local:8200"
+      path: "secret"
+      version: "v2"
+      auth:
+        tokenSecretRef:
+          name: "vault-token"
+          key: "token"
+
```
Comment (saraswatpuneet) [inline deployment/k8s/vault/vault_secret_store.yml:15]: external secrets is bunch of CRs that enable integration with various backend, here we are setting our provider to vault and provide basic auth that will allow this store to sync secrets from vault

File: deployment/k8s/frequency-gateway/values.yaml [hunk 2]
```diff
@@ -53,6 +53,12 @@ nodeSelector: {}
 tolerations: []
 affinity: {}
 
+# Support vault integration
+vault:
+  enabled: false 
+  secretsPath: "secret/frequency-gateway"
+  role: "frequency-gateway-role"
+
```
Comment (saraswatpuneet) [inline deployment/k8s/frequency-gateway/values.yaml:61]: Assuming a secret store with vault is created and deployment will pull secrets from the store , currently using basic auth so role isnt used. But these CRDs offer various auth mechanism between provider and secret store

File: deployment/k8s/frequency-gateway/templates/vault_secrets_hooks.yml [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    - secretKey: PROVIDER_ACCOUNT_SEED_PHRASE
+      remoteRef:
+        key: "{{ .Values.vault.secretsPath }}/graph"
+        property: PROVIDER_ACCOUNT_SEED_PHRASE
+    - secretKey: PROVIDER_ACCESS_TOKEN
+      remoteRef:
+        key: "{{ .Values.vault.secretsPath }}/graph"
+        property: PROVIDER_ACCESS_TOKEN
+
+{{- end }}
```
Comment (saraswatpuneet) [inline deployment/k8s/frequency-gateway/templates/vault_secrets_hooks.yml:98]: All the external secrets from secret store are deployed in format deployment is expecting

File: deployment/k8s/frequency-gateway/templates/vault_secrets_hooks.yml [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+spec:
+  refreshInterval: "15s" 
+  secretStoreRef:
+    name: vault-backend
+    kind: SecretStore
+  target:
+    name: graph-secret
+    creationPolicy: Owner
+  data:
+    - secretKey: PROVIDER_ACCOUNT_SEED_PHRASE
```
Comment (aramikm) [inline deployment/k8s/frequency-gateway/templates/vault_secrets_hooks.yml:89]: We also need AT_REST_ENCRYPTION_KEY_SEED

File: docs/src/Run/Vault.md [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+- **Content Watcher Service:**
+
+  ```bash
+  vault kv put secret/frequency-gateway/content-watcher IPFS_BASIC_AUTH_USER=<username> IPFS_BASIC_AUTH_SECRET=<password>
+  ```
+
+- **Graph Service:**
+
+  ```bash
+  vault kv put secret/frequency-gateway/graph PROVIDER_ACCESS_TOKEN=<your-access-token> PROVIDER_ACCOUNT_SEED_PHRASE=<your-seed-phrase>
```
Comment (aramikm) [inline docs/src/Run/Vault.md:79]: also needs `AT_REST_ENCRYPTION_KEY_SEED`

File: deployment/k8s/frequency-gateway/templates/vault_secrets_hooks.yml (hunk 4 above)
Comment (saraswatpuneet) [reply deployment/k8s/frequency-gateway/templates/vault_secrets_hooks.yml:89]: We can do that in separate PR but thanks for reminder

**Pull request merged**